zfit/models/physics.py

```diff
@@ -548,3 +548,400 @@ class GeneralizedCBPDFRepr(BasePDFRepr):
+               model. This will be used as the default space of the PDF and,
+               if not given explicitly, as the normalization range.
+
+               The default space is used for example in the sample method: if no
+               sampling limits are given, the default space is used.
+
+               The observables are not equal to the domain as it does not restrict or
+               truncate the model outside this range. |@docend:pdf.init.obs|
+            extended: |@doc:pdf.init.extended| The overall yield of the PDF.
+               If this is parameter-like, it will be used as the yield,
+               the expected number of events, and the PDF will be extended.
+               An extended PDF has additional functionality, such as the
+               ``ext_*`` methods and the ``counts`` (for binned PDFs). |@docend:pdf.init.extended|
+            norm: |@doc:pdf.init.norm| Normalization of the PDF.
+               By default, this is the same as the default space of the PDF. |@docend:pdf.init.norm|
+            name: |@doc:pdf.init.name| Human-readable name
+               or label of
+               the PDF for better identification.
+               Has no programmatical functional purpose as identification. |@docend:pdf.init.name|
+        """
+        params = {
+            "mu": mu,
+            "sigmal": sigmal,
+            "alphal": alphal,
+            "nl": nl,
+            "sigmar": sigmar,
+            "alphar": alphar,
+            "nr": nr,
+        }
+        super().__init__(obs=obs, name=name, params=params, extended=extended, norm=norm)
+
+    def _unnormalized_pdf(self, x):
+        mu = self.params["mu"].value()
+        sigmal = self.params["sigmal"].value()
+        alphal = self.params["alphal"].value()
+        sigmar = self.params["sigmar"].value()
+        nl = self.params["nl"].value()
+        alphar = self.params["alphar"].value()
+        nr = self.params["nr"].value()
+        x = z.unstack_x(x)
+        return generalized_gaussexptail_func(
+            x=x,
+            mu=mu,
+            sigmal=sigmal,
+            alphal=alphal,
+            sigmar=sigmar,
+            nl=nl,
+            alphar=alphar,
+            nr=nr,
+        )
+
+
+class GeneralizedGaussExpTailPDFRepr(BasePDFRepr):
+    _implementation = GeneralizedGaussExpTail
+    hs3_type: Literal["GeneralizedGaussExpTail"] = pydantic.Field("GeneralizedGaussExpTail", alias="type")
+    x: SpaceRepr
+    mu: Serializer.types.ParamTypeDiscriminated
+    sigmal: Serializer.types.ParamTypeDiscriminated
+    alphal: Serializer.types.ParamTypeDiscriminated
+    sigmar: Serializer.types.ParamTypeDiscriminated
+    nl: Serializer.types.ParamTypeDiscriminated
+    alphar: Serializer.types.ParamTypeDiscriminated
+    nr: Serializer.types.ParamTypeDiscriminated
+
+
+GeneralizedGaussExpTail.register_analytic_integral(
+    func=generalized_gaussexptail_integral, limits=gaussexptail_integral_limits
+)
+
+
+class CBExGauss:
```

Comment:
This requires some more thoughts I think. I would tendentially not follow the naming scheme and behavior used in the root implementation: "tailleft" is not very explanatory, the choices are highly arbitrary: why is the left tail GaussExp, and if it's a right one a CB?

We should probably start with: what does a user need? Maybe a one-tailed PDF with two sigmas, could be. In this case, it makes sense to be able to deactivate one tail, one _could_ know or read in the docs how to do it, i.e. set a high value, but if we do it, could also work.

If they want to mix both tales, that's another task (but is that needed?), and maybe a piecewise PDF could do the job, a more generalized one. Do you think the mixing of tales is needed?

For leaving away a tail, what about we allow the user to set parameters to `False`, i.e. either both left or right tail parameters in the generalized versions? This should be unambiguous. We could either set the limits high, but that's maybe a bit surprising for a user that the param is there, or simply not set them and adjust the (integral-)functions for the case that these parameters are not present in `params`.

Or what do you think, any specific use-cases in mind?

We could also just do nothing and have it in the docs?

Reply:
Yes, actually let's not do the factory for now. Can I leave it commented out for future reference?

Reply:
Sure, just leave a comment why it has been commented out, but that's fine